### PR TITLE
[WIP] interim guidance on deploying to production environments

### PR DESCRIPTION
**Pull request by geoffcline**
**1. Issue, if available:**
#815 

**2. Description of changes:**
add page including a "checklist" or description of requirements for installing karpenter on an existing (perhaps production) cluster

**3. Does this change impact docs?**
- [ x ] Yes, PR includes docs updates
- [ ] Yes, issue opened: *link to issue*
- [ ] No

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by geoffcline** [comment]
@chrisnegus let me know what you are thinking about the ~style~ of this guide so far. It's turned out to be more of a discussion than a listing of requirements.

**Comment by geoffcline** [comment]
page for review: https://deploy-preview-838--karpenter-docs-prod.netlify.app/pre-docs/deploy-guide/

File: website/content/en/pre-docs/deploy-guide/_index.md [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+
+### Minimum Kubernetes Version
+
+Karpenter supports the [same set of Kubernetes versions as EKS supports](https://docs.aws.amazon.com/eks/latest/userguide/kubernetes-versions.html) for new clusters. Karpenter strongly encourages a posture of promptly adopting new versions of Kubernetes.
+
+### Service Account
+    - `Karpenter` Service Account
+
+The `Karpenter` service account needs sufficient cluster permissions to, for example, add new nodes and bind pods to nodes. Review the ["karpenter-controller" `role` and `ClusterRole`](rbac.yaml) from the default helm chart to understand what permissions the Karpenter service account needs. The Karpenter helm chart includes other roles (`karpenter-webhook`), but the "karpenter-controller" resources require uniquely broad access.
```
Comment (ellistarn) [inline website/content/en/pre-docs/deploy-guide/_index.md:30]: Let's point to https://github.com/awslabs/karpenter/blob/main/charts/karpenter/templates/controller/rbac.yaml to avoid needing to update the file in multiple places.

File: website/content/en/docs/launch-templates.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
+# Custom Images
```
Comment (geoffcline) [inline website/content/en/docs/launch-templates.md:1]: oh no this should not be here. please hold.

File: website/content/en/pre-docs/deploy-guide/_index.md [hunk 3]
```diff
@@ -0,0 +1,165 @@
+---
+title: "Deployment Guide"
+linkTitle: "Deployment Guide"
+weight: 50
+---
+
+Introducing new software the modifies how Kubernetes operates is complex, and no single solution exists.
```
Comment (ellistarn) [inline website/content/en/pre-docs/deploy-guide/_index.md:7]: This is self evident from the remainder of this doc.
```suggestion
```

File: website/content/en/pre-docs/deploy-guide/_index.md [hunk 4]
```diff
@@ -0,0 +1,165 @@
+---
```
Comment (ellistarn) [inline website/content/en/pre-docs/deploy-guide/_index.md:1]: Why is the path to this file and the title of this file *slightly* different? Can we align around /deployment-guide/ and Deployment Guide

**Comment by geoffcline** [comment]
the only interesting file is

website/content/en/pre-docs/deploy-guide/_index.md

the other files are merging changes made to /docs/ into /pre-docs/

File: website/content/en/pre-docs/concepts/_index.md [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+
+Users fall under two basic roles: Kubernetes cluster operators and application developers.
+This document describes Karpenter concepts through the lens of those two types of users.
+
+## Cluster operator
+
+As a Kubernetes cluster operator, you can engage with Karpenter to:
+
+* Install Karpenter 
+* Configure provisioners to set constraints and other features for managing nodes
```
Comment (rothgar) [inline website/content/en/pre-docs/concepts/_index.md:15]:
```suggestion
* Configure provisioners
```

File: website/content/en/pre-docs/concepts/_index.md [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+* Deprovision nodes
+* Upgrade nodes
+
+Concepts associated with this role are described below.
+
+
+### Installing Karpenter
+
+Karpenter is designed to run on a node in your Kubernetes cluster.
+As part of the installation process, you need credentials from the underlying cloud provider to allow nodes to be started up and added to the cluster as they are needed.
```
Comment (rothgar) [inline website/content/en/pre-docs/concepts/_index.md:25]:
```suggestion
As part of the installation process, you need credentials (e.g. IAM) from your infrastructure provider to allow nodes to be started up and added to the cluster as they are needed.
```

File: website/content/en/pre-docs/concepts/_index.md [hunk 7]
```diff
@@ -0,0 +1,167 @@
...
+
+[Getting Started with Karpenter on AWS](https://karpenter.sh/docs/getting-started/)
+describes the process of installing Karpenter on an AWS cloud provider.
+Because requests to add and delete nodes and schedule pods are made through Kubernetes, AWS IAM Roles for Service Accounts (IRSA) are needed by your Kubernetes cluster to make privileged requests to AWS.
```
Comment (rothgar) [inline website/content/en/pre-docs/concepts/_index.md:29]: IRSA isn't a requirement but is recommended. Lots of people still use kube2iam or similar tools.

File: website/content/en/pre-docs/concepts/_index.md [hunk 8]
```diff
@@ -0,0 +1,167 @@
...
+For example, Karpenter uses AWS IRSA roles to grant the permissions needed to describe EC2 instance types and create EC2 instances.
```
Comment (rothgar) [inline website/content/en/pre-docs/concepts/_index.md:30]: This is a bit confusing because Karpenter doesn't use IRSA. Karpenter uses a Kubernetes service account mounted in the pod. How that service account get's access into AWS depends on the implementation.

File: website/content/en/pre-docs/concepts/_index.md [hunk 9]
```diff
@@ -0,0 +1,167 @@
...
+
+Once privileges are in place, Karpenter is deployed with a Helm chart.
```
Comment (rothgar) [inline website/content/en/pre-docs/concepts/_index.md:32]:
```suggestion
Once privileges are in place, Karpenter can be deployed with a Helm chart.
```

File: website/content/en/pre-docs/concepts/_index.md [hunk 10]
```diff
@@ -0,0 +1,167 @@
...
+
+Karpenter's job is to add nodes to handle unschedulable pods, schedule pods on those nodes, and remove the nodes when they are not needed.
+To configure Karpenter, you create *provisioners* that define how Karpenter manages unschedulable pods and expires nodes.
+Here are some things to know about the Karpenter provisioner:
+
+* **Unschedulable pods**: Karpenter only attempts to provision pods that have a status condition `Unschedulable=True`, which the kube scheduler sets when it fails to schedule the pod to existing capacity.
+
+* **Provisioner CR**: Karpenter defines a Custom Resource called a Provisioner to specify provisioning configuration.
+Each provisioner manages a distinct set of nodes, but pods can be scheduled to any provisioner that supports its scheduling constraints.
+A provisioner contains constraints that impact the nodes that can be provisioned and attributes of those nodes (such timers for removing nodes).
```
Comment (rothgar) [inline website/content/en/pre-docs/concepts/_index.md:44]:
```suggestion
A provisioner contains constraints that impact the nodes that can be provisioned and attributes of those nodes (such as timers for removing nodes).
```

File: website/content/en/pre-docs/concepts/_index.md [hunk 11]
```diff
@@ -0,0 +1,167 @@
...
+See [Provisioner](/docs/provisioner-crd/) for a description of settings and the [Provisioning](/docs/tasks/provisioner.md) task for of provisioner examples. 
+
+* **Well-known labels**: The provisioner can use well-known Kubernetes labels to allow pods to request only certain instance types, architectures, operating systems, or other attributes when creating nodes.
+See [Well-Known Labels, Annotations and Taints](https://kubernetes.io/docs/reference/labels-annotations-taints/) for details.
+Keep in mind that only a subset of these labels are supported in Karpenter, as described later.
```
Comment (rothgar) [inline website/content/en/pre-docs/concepts/_index.md:49]:
```suggestion
Keep in mind that only a subset of these labels are supported in Karpenter, as described in the [application developer section]({{< ref "#application-developer" >}}).
```

File: website/content/en/pre-docs/concepts/_index.md [hunk 12]
```diff
@@ -0,0 +1,167 @@
...
+
+* **Deprovisioning nodes**: A provisioner can also include time-to-live values to indicate when nodes should be deprovisioned after a set amount of time from when they were created or after they becomes empty of deployed pods.
```
Comment (rothgar) [inline website/content/en/pre-docs/concepts/_index.md:51]:
```suggestion
* **Deprovisioning nodes**: A provisioner can also include time-to-live (TTL) values to indicate when nodes should be deprovisioned after a set amount of time from when they were created or after they becomes empty of deployed pods.
```

Does TTL affect both total time a node can exist and how long it should stay after empty? I thought TTL only affected total life of a node before it would be destroyed.

File: website/content/en/pre-docs/concepts/_index.md [hunk 13]
```diff
@@ -0,0 +1,167 @@
...
+
+* **Multiple provisioners**: Multiple provisioners can be configured on the same cluster.
+For example, you might want to configure different teams on the same cluster to run on completely separate capacity.
+One team could run on nodes nodes using BottleRocket, while another uses EKSOptimizedAMI.
```
Comment (rothgar) [inline website/content/en/pre-docs/concepts/_index.md:55]: Is the R in bottlerocket usually capitalized? All the branding I see is lower case but maybe the SSM search path is capitalized.

File: website/content/en/pre-docs/concepts/_index.md [hunk 14]
```diff
@@ -0,0 +1,167 @@
...
+
+Although most use cases are addressed with a single provisioner for multiple teams, multiple provisioners are useful to isolate nodes for billing, use different node constraints (such as no GPUs for a team), or use different deprovisioning settings.
+
+### Deprovisioning nodes
+
+Karpenter deletes nodes when they are no longer needed.
+
+* **Finalizer**: Karpenter places a finalizer bit on each node it creates.
+When a request comes in to delete one of those nodes (such as a TTL or a manual `kubectl delete node`), Karpenter will cordon the node, drain all the pods, terminate the EC2 instance, and delete the node object.
+Karpenter handles all clean-up work needed to properly delete the node.
```
Comment (rothgar) [inline website/content/en/pre-docs/concepts/_index.md:65]: Including EBS volumes? If so we should add that because it's a question we get asked.

File: website/content/en/pre-docs/concepts/_index.md (hunk 12 above)
Comment (rothgar) [reply website/content/en/pre-docs/concepts/_index.md:51]: I see my answer further below and realized you said "values" and not "value" which answers my question.

File: website/content/en/pre-docs/concepts/_index.md [hunk 15]
```diff
@@ -0,0 +1,167 @@
...
+* **Node Expiry**: If a node expiry time-to-live value (`ttlSecondsUntilExpired`) is reached, that node is drained of pods and deleted (even if it is still running workloads).
+* **Empty nodes**: When the last workload pod running on a Karpenter-managed node is gone, the node is annotated with an emptiness timestamp.
+Once that "node empty" time-to-live (`ttlSecondsAfterEmpty`) is reached, finalization is triggered.
+
+For more details on how Karpenter deletes nodes, see [Deleting nodes with Karpenter](/docs/tasks/delete-nodes.md) for details.
+
+### Upgrading nodes
+
+A straight-forward way to upgrade nodes is to set `ttlSecondsUntilExpired`.
+Nodes will be terminated after a set period of time and will be replaced with newer nodes.
```
Comment (rothgar) [inline website/content/en/pre-docs/concepts/_index.md:75]: Are there any safeguards against multiple nodes being terminated at once? If all my nodes are created at the same time will they all be destroyed at the same time too? Do we have a way to add jitter or skew to this TTL or locking to make sure the entire cluster doesn't recycle at once?

File: website/content/en/pre-docs/concepts/_index.md [hunk 16]
```diff
@@ -0,0 +1,167 @@
...
+
+For details on upgrading nodes with Karpenter, see [Upgrading nodes with Karpenter](/docs/tasks/upgrade-nodes.md) for details.
+
+
+Understanding the following concepts will help you in carrying out the tasks just described.
+
+### Constraints
+
+The concept of layered constraints is key to using Karpenter.
+With no constraints defined in provisioners and none requested from pods being deployed, Karpenter chooses from the entire universe of features available to your cloud provider.
```
Comment (rothgar) [inline website/content/en/pre-docs/concepts/_index.md:85]:
```suggestion
With no constraints defined in provisioners and none requested from pods being deployed, Karpenter chooses from all available compute types that satisfy the pods requirements available in to your provider.
```

File: website/content/en/pre-docs/deploy-guide/_index.md [hunk 17]
```diff
@@ -0,0 +1,165 @@
...
+
+Introducing new software the modifies how Kubernetes operates is complex, and no single solution exists. 
+
+This guide describes how Karpenter interacts with external resources, and modifies cluster behavior. 
+
+## Resources Dependencies
+
+### Cluster on AWS
+
+Currently, Karpenter creates nodes (EC2 Instances) through the AWS cloud provider module, the only such module implemented. Clusters on other cloud providers are contemplated, but are not currently supported.
```
Comment (rothgar) [inline website/content/en/pre-docs/deploy-guide/_index.md:15]:
```suggestion
Karpenter creates nodes (EC2 Instances) through the AWS cloud provider, the only such module implemented.
Node provisioning on other providers is not currently supported.
If you would like to use Karpenter with other infrastructure providers please open an issue to let us know your use case.
```
Do we call it a "module" other places. I think for users they don't need to know the terminology in the code

File: website/content/en/pre-docs/deploy-guide/_index.md [hunk 18]
```diff
@@ -0,0 +1,165 @@
...
+
+## Resources Dependencies
+
+### Cluster on AWS
+
+Currently, Karpenter creates nodes (EC2 Instances) through the AWS cloud provider module, the only such module implemented. Clusters on other cloud providers are contemplated, but are not currently supported. 
+
+Theoretically, a control plane could be hosted separately from Karpenter provisioned AWS instances. However, this complicates identity and authorization substantially. The Karpenter controller (i.e., agent) would need authorization with AWS APIs to provision new instances, and permissions with the control plane to add the new nodes. 
+
+Additionally, the intricacies of securing control plane communication between clouds presents another significant problem.
```
Comment (rothgar) [inline website/content/en/pre-docs/deploy-guide/_index.md:19]: This is a bit wordy and since we already tell people basically to "don't do this" couldn't we have this be a sentence that says.

>It is recommended to run Karpenter in the cluster where it will be managing nodes. This simplifies authentication, communication, and troubleshooting for cluster administrators.

File: website/content/en/pre-docs/deploy-guide/_index.md [hunk 19]
```diff
@@ -0,0 +1,165 @@
...
+### Cluster on AWS
+
+Currently, Karpenter creates nodes (EC2 Instances) through the AWS cloud provider module, the only such module implemented. Clusters on other cloud providers are contemplated, but are not currently supported. 
+
+Theoretically, a control plane could be hosted separately from Karpenter provisioned AWS instances. However, this complicates identity and authorization substantially. The Karpenter controller (i.e., agent) would need authorization with AWS APIs to provision new instances, and permissions with the control plane to add the new nodes. 
+
+Additionally, the intricacies of securing control plane communication between clouds presents another significant problem. 
+
+The getting started guide covers [configuring EKS to accept newly provisioned instances as nodes](../getting-started/#create-the-karpenternode-iam-role) (see, *KarpenterNodeRole* and *system:bootstrappers*). If using a self-managed (or alternatively provided) control plane, this must be manually handled. Consult the [*User Data - Autoconfigure* section](../cloud-providers/aws/launch-templates/#user-data---autoconfigure) of *Launch Templates* for more information on joining nodes to a cluster.
```
Comment (rothgar) [inline website/content/en/pre-docs/deploy-guide/_index.md:21]: In the second sentence what does "alternatively provided" mean? Isn't that the same as self-managed?

File: website/content/en/pre-docs/deploy-guide/_index.md [hunk 20]
```diff
@@ -0,0 +1,165 @@
...
+
+### Minimum Kubernetes Version
+
+Karpenter supports the [same set of Kubernetes versions as EKS supports](https://docs.aws.amazon.com/eks/latest/userguide/kubernetes-versions.html) for new clusters. Karpenter strongly encourages a posture of promptly adopting new versions of Kubernetes.
```
Comment (rothgar) [inline website/content/en/pre-docs/deploy-guide/_index.md:25]:
```suggestion
Karpenter supports the [same versions of Kubernetes as EKS](https://docs.aws.amazon.com/eks/latest/userguide/kubernetes-versions.html).
You can also use Karpenter with non-EKS clusters so long as you configure the correct IAM permissions and use a supported Kubernetes version.
Karpenter strongly encourages a posture of promptly adopting new versions of Kubernetes as new constraints and node features are added regularly.

```

File: website/content/en/pre-docs/deploy-guide/_index.md [hunk 21]
```diff
@@ -0,0 +1,165 @@
...
+  - create
+[...]
+```
+
+
+### IAM Roles
+- `Karpenter` Role
+- `KarpenterNode` Role
+
+Generally, Karpenter needs to authenticate with the AWS EC2 API to create new instances. This is best done with IRSA, IAM Roles for Service Accounts. IRSA naturally extends the `Karpenter` service account to handle interacting with AWS APIs. The getting started guide describes creating a cluster service account mapped to an IAM role with sufficient permissions.
```
Comment (rothgar) [inline website/content/en/pre-docs/deploy-guide/_index.md:55]:
```suggestion
Karpenter needs to authenticate with the AWS EC2 API to create and delete instances.
It is recommended to do this with [IAM Roles for Service Accounts (IRSA)](https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html).
The getting started guide describes creating a cluster service account mapped to an IAM role with sufficient permissions. 

```
The third paragraph was confusing from someone not familiar with IRSA. I don't think it's necessary to call out.

File: website/content/en/pre-docs/deploy-guide/_index.md [hunk 22]
```diff
@@ -0,0 +1,165 @@
...
+```
+
+
+### IAM Roles
+- `Karpenter` Role
+- `KarpenterNode` Role
+
+Generally, Karpenter needs to authenticate with the AWS EC2 API to create new instances. This is best done with IRSA, IAM Roles for Service Accounts. IRSA naturally extends the `Karpenter` service account to handle interacting with AWS APIs. The getting started guide describes creating a cluster service account mapped to an IAM role with sufficient permissions. 
+
+Karpenter may alternatively be configured to authenticate with AWS using the `kube2iam` package, or with locally stored Access Keys. Both practices are discouraged.
```
Comment (rothgar) [inline website/content/en/pre-docs/deploy-guide/_index.md:57]: Why are they discouraged?

File: website/content/en/pre-docs/deploy-guide/_index.md [hunk 23]
```diff
@@ -0,0 +1,165 @@
...
+
+### IAM Roles
+- `Karpenter` Role
+- `KarpenterNode` Role
+
+Generally, Karpenter needs to authenticate with the AWS EC2 API to create new instances. This is best done with IRSA, IAM Roles for Service Accounts. IRSA naturally extends the `Karpenter` service account to handle interacting with AWS APIs. The getting started guide describes creating a cluster service account mapped to an IAM role with sufficient permissions. 
+
+Karpenter may alternatively be configured to authenticate with AWS using the `kube2iam` package, or with locally stored Access Keys. Both practices are discouraged. 
+
+The provisioned instances must comparable permissions to a standard EKS self-managed node. For example, permission to configure instance networking interfaces. The getting started guide again provides a useful starting point in [the cloudformation template](../getting-started/cloudformation.yaml).
```
Comment (rothgar) [inline website/content/en/pre-docs/deploy-guide/_index.md:59]:
```suggestion
The provisioned instances must comparable permissions to a standard EKS self-managed node. For example, permission to configure instance networking interfaces. The getting started guide again provides a useful starting point in [the CloudFormation template](../getting-started/cloudformation.yaml).

```

File: website/content/en/pre-docs/deploy-guide/_index.md [hunk 24]
```diff
@@ -0,0 +1,165 @@
...
+
+The tags must have this form:
+- key: `kubernetes.io/cluster/${CLUSTER_NAME}`
+- value: "" (empty string)
+
+In some situations, EKS may automatically apply this tag to subnets. Do not rely on EKS to do this. 
+
+### `eksctl`
+
+The getting started guide uses `eksctl`. This tool is officially supported and sponsored by AWS. However, equivalent EKS clusters may be created using CloudFormation, or even the web console.
```
Comment (rothgar) [inline website/content/en/pre-docs/deploy-guide/_index.md:73]:
```suggestion
The getting started guide uses `eksctl`. This tool is officially supported and sponsored by AWS. However, equivalent EKS clusters may be created using CloudFormation, Terraform, or even the web console. 

```
A majority of our users use terraform for cluster creation so we should call it out

File: website/content/en/pre-docs/deploy-guide/_index.md [hunk 25]
```diff
@@ -0,0 +1,165 @@
...
+- key: `kubernetes.io/cluster/${CLUSTER_NAME}`
+- value: "" (empty string)
+
+In some situations, EKS may automatically apply this tag to subnets. Do not rely on EKS to do this. 
+
+### `eksctl`
+
+The getting started guide uses `eksctl`. This tool is officially supported and sponsored by AWS. However, equivalent EKS clusters may be created using CloudFormation, or even the web console. 
+
+The ["cluster.yaml" file](../getting-started/#create-a-cluster) provides does succinctly describe a suitable EKS cluster for Karpenter, including an OIDC provider for IAM (the infrastructure for IRSA).
```
Comment (rothgar) [inline website/content/en/pre-docs/deploy-guide/_index.md:75]:
```suggestion
The ["cluster.yaml" file](../getting-started/#create-a-cluster) provides a suitable `eksctl` cluster configuration for Karpenter, including an OIDC provider for IAM (a requirement for IRSA). 

```

Comment (rothgar) [reply website/content/en/pre-docs/deploy-guide/_index.md:75]: Should we show an example `eksctl` cluster create command here?

File: website/content/en/pre-docs/deploy-guide/_index.md [hunk 26]
```diff
@@ -0,0 +1,165 @@
...
+```
+
+
+### Launch Template
+
+A launch template is an AWS resource that bundles every configuration value needed to properly launch a new instance, such as base image and storage.
+
+Karpenter will create a launch template suitable for the environment described in the getting started guide (EKS, IRSA, etc) if none is specified.
+
+To the extent possible, nodes should be customized using Karpenter provisioner CRD fields. However, many environments require using a different base image (instead of the EKS optimized flavor of Amazon Linux 2), such as to incorporate a standardized security daemon at the node level.
```
Comment (rothgar) [inline website/content/en/pre-docs/deploy-guide/_index.md:102]: Isn't Bottlerocket the default?

File: website/content/en/pre-docs/deploy-guide/_index.md [hunk 27]
```diff
@@ -0,0 +1,165 @@
...
+
+Karpenter will create a launch template suitable for the environment described in the getting started guide (EKS, IRSA, etc) if none is specified.
+
+To the extent possible, nodes should be customized using Karpenter provisioner CRD fields. However, many environments require using a different base image (instead of the EKS optimized flavor of Amazon Linux 2), such as to incorporate a standardized security daemon at the node level. 
+
+Review the [detailed instructions](../cloud-providers/aws/launch-templates/) for creating a custom launch template. 
+
+## Deploy Karpenter
+
+The yaml files comprising the helm chart are suitable for production environments. They may be lightly modified for direct use, for example storage in an internal GitOps cluster repo (without referencing an external chart).
```
Comment (rothgar) [inline website/content/en/pre-docs/deploy-guide/_index.md:108]: >They may be lightly modified for direct use, for example storage in an internal GitOps cluster repo (without referencing an external chart).

This was confusing for me and the call-out to GitOps seems out of place IMO.

File: website/content/en/pre-docs/deploy-guide/_index.md [hunk 28]
```diff
@@ -0,0 +1,165 @@
...
+Some important fields are:
+- serviceAccount.create
+- serviceAccount.name
+- controller.clusterName
+- controller.clusterEndpoint
+
+## Mutations of Kubernetes Behavior
+Note: This section is an incomplete draft.
+
+- `k delete node`
```
Comment (rothgar) [inline website/content/en/pre-docs/deploy-guide/_index.md:136]:
```suggestion
- `kubectl delete node`

```

File: website/content/en/pre-docs/deploy-guide/_index.md [hunk 29]
```diff
@@ -0,0 +1,165 @@
...
+    - Deleting a karpenter managed node triggers deprovisioning of the cloud instance.
+- High amount of pods on a single node
+    - Karpenter often decides to use a large instance, and pack many pods onto the instance. 
+    - This may increase the load on *per node* daemonsets
```
Comment (rothgar) [inline website/content/en/pre-docs/deploy-guide/_index.md:140]: It also can reduce the load on the Kubernetes API and may need VPC CNI settings (eg prefix delegation). Do we have a doc somewhere we can reference with trade-offs and considerations?

File: website/content/en/pre-docs/deploy-guide/_index.md (hunk 19 above)
Comment (geoffcline) [reply website/content/en/pre-docs/deploy-guide/_index.md:23]: is there such a thing as a 3rd party managed control plane on AWS? ROSA perhaps? Regardless, I just struck "alternatively provided"

File: website/content/en/pre-docs/deploy-guide/_index.md (hunk 1 above)
Comment (geoffcline) [reply website/content/en/pre-docs/deploy-guide/_index.md:35]: agreed

File: website/content/en/pre-docs/deploy-guide/_index.md (hunk 22 above)
Comment (geoffcline) [reply website/content/en/pre-docs/deploy-guide/_index.md:65]: added a sentence describing this. I'm under the impression that the usual response to "kube2iam" is "people still use that?" -- although I'm not sure if the attitude is the same outside amazon.

File: website/content/en/pre-docs/deploy-guide/_index.md (hunk 26 above)
Comment (geoffcline) [reply website/content/en/pre-docs/deploy-guide/_index.md:113]: not anymore :)

File: website/content/en/pre-docs/deploy-guide/_index.md (hunk 27 above)
Comment (geoffcline) [reply website/content/en/pre-docs/deploy-guide/_index.md:119]: good point. revised.

File: website/content/en/pre-docs/deploy-guide/_index.md (hunk 29 above)
Comment (geoffcline) [reply website/content/en/pre-docs/deploy-guide/_index.md:152]: added those additional points. i'll take a look for existing documents.

File: website/content/en/pre-docs/deploy-guide/_index.md (hunk 3 above)
Comment (geoffcline) [reply website/content/en/pre-docs/deploy-guide/_index.md:7]: This statement explicitly sets a mindset I want users to approach this doc with.

File: website/content/en/pre-docs/deploy-guide/_index.md (hunk 4 above)
Comment (geoffcline) [reply website/content/en/pre-docs/deploy-guide/_index.md:1]: is is "deploy" because I kept reading "deployment" as "development" and vice versa. I can change it now.

**Comment by geoffcline** [comment]
I revised the PR to exclude the extraneous files. They are now handled by #844 

I will store Justin's feedback on the concepts page.

**Comment by geoffcline** [comment]
- reverting to draft status
- concerns about scope
- concerns about "sweeping statements" re aws/eks

- may defer to after launch
- or, reduce to MVP (e.g., reference list of IAM policy templates)

File: website/content/en/pre-docs/deployment-guide/_index.md [hunk 30]
```diff
@@ -0,0 +1,182 @@
...
+title: "Deployment Guide"
+linkTitle: "Deployment Guide"
+weight: 50
+---
+
+Introducing new software the modifies how Kubernetes operates is complex, and no single solution exists. 
+
+This guide describes how Karpenter interacts with external resources, and modifies cluster behavior. 
+
+## Resource Dependencies
```
Comment (geoffcline) [inline website/content/en/pre-docs/deployment-guide/_index.md:11]: add section on {subnets, security groups, network interfaces}
